### ISSUE TITLE
Advise in Reinforcement Learning for very long horizon

### ISSUE THREAD

**Issue by ldesmet** [DLR-RM/stable-baselines3#939]
Hello everyone,

I am working on a RL project, and I am looking for some help/advise as I did not reached singificant good results. Here is the description of the problem.

 - Global Description
The agent has to follow the always same path. The path is composed of 10000 steps and **the agent always has to go further** (it can't go backward or stay in the same state) and will always reach final state after these 10000 steps. 
At each step, an amount of energy is requested or given by the environment. The decision to take is how much energy the agent is willing to give / accept. This decision will have an impact on the value of a specific component (let's call it X). The more energy is given, **the more value of X will decrease** and vice versa.
At the begining X has an init value of 0.3 and should finish at 0.7.

 - The environment state is composed of 3 float:
      1. Position of the agent in the path **[values: 0.0 to 1.0]** *(0-> begining; 1-> final state/end of path)*
      2. Value X of the specific component **[values: 0.0 to 1.0]** *(initialization state -> X=0.3; and final state X should be equal to 0.7)*
     3. The normalized energy the environment requires / give **[values: -1.0 to 1.0]** (when negative, the environment gives while when it is positive, it requires)

 - Description of Agent:
For now I have used PPO agent. It takes as input those 3 floats above and will output a unique float between 0 and 1 which indicates how much energy it will match (0-> no match at all; and 1-> fully match what environment gives/erquires)

 - Description of Penalities:
At final state, the final negative reward is the absolute difference between X value and target (0.7) times a factor of -1000. 
At each step, if the agent does not match the energy requested, it will receive a negative reward proportional to the amount of missing energy (typically value is around -0.5...). Last but not least, value of X is free (can potentially reached negative value or above 1). In this case a huge penality is provided following this equation *R = -1000 * X*(X-1)

 - Framework and algorithm
I m using stable baseline 3 with PPO agent with following hyperparameter :

```
model = PPO(
    policy='MlpPolicy',
    env=custom_env,
    learning_rate=0.01,
    n_steps = 2048, # How much steps before training again
    batch_size=1024,
    n_epochs = 5,
    gamma=0.99,
    gae_lambda = 0.95,
    clip_range = 0.2,
    normalize_advantage = True,
    ent_coef = 0.01,
    vf_coef=0.01,
    target_kl = 0.01,
    tensorboard_log=log_path,
    verbose=1)
```

Here is the current results I am getting

[![Tensorboard log][1]][1]

My list of questions:

 - Do you think PPO is well adressed for this kind of problem ?
 - Why do I have such variance ? Why my agent seems to not converge ?
 - Should I play with my reward values or should I play with hyperparamaters ?
 - I don't know if I don't wait long enough or if my algorithm will never converge

Any comments / avise would be greatly welcome.

Thanks a lot !

  [1]: https://i.stack.imgur.com/dwQjW.png

**Comment by Miffyli** [comment]
Hey. Sadly we do not have time to offer tech support. See the links [here](https://github.com/DLR-RM/stable-baselines3/blob/master/.github/ISSUE_TEMPLATE/question.md) for other forums to ask this question in.

Closing as no tech support.
